### ISSUE TITLE
[Feature]: Popover without any focusable item inside is not a good a11y pattern

### ISSUE THREAD

**Issue by YuanboXue-Amber** [microsoft/fluentui#26916]
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

When v9 PopoverSurface has focusable item, the focus lands inside it on open. 
However when PopoverSurface has no focusable item, focus stays on trigger on open. This is a questionable a11y pattern: screen reader user doesn't know how to get inside PopoverSurface, except when it is inline popover.

We are using examples like this in our docsite: https://react.fluentui.dev/?path=/docs/components-popover--default

I purpose to add a check [at where popover is trying to find focusable items inside PopoverSurface](https://github.com/microsoft/fluentui/blob/708575027e2b6b06d778f719c2ef5a53fe4c762c/packages/react-components/react-popover/src/components/Popover/usePopover.ts#L129). If there's no focusable, and no `inline` prop, then console warn the user.
And also we should correct our examples.

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by YuanboXue-Amber** [comment]
Discussed this with @ling1726 , maybe we can focus on PopoverSurface for a non-inline popover without any focusable items. Dialog focuses on DialogSurface when there's no focusable items inside.
